Document GraphQL client errors (#973)
getsentry/develop: src/docs/sdk/features.mdx

- data (all fields are optional but recommended):
  - `url` - The URL used in the HTTP request
  - `method` - uppercase HTTP method, i.e: GET, HEAD

>   - `status_code` - Numeric status code such as `200` or `404`

Author: Not sure, by mistake, good catch.

As an example, see the [OkHTTP Client integration](https://github.com/getsentry/sentry-java/pull/2287) for Android.

## GraphQL Client Integrations

The GraphQL Client integrations match the [HTTP Client Integrations](/sdk/features/#http-client-integrations) with a few differences:

> The `failedRequestStatusCodes` parameter does not exist because GraphQL errors are not HTTP errors, sometimes the request is errored and the response is successful.

Reviewer: ```suggestion
The `failedRequestStatusCodes` parameter does not exist because GraphQL errors are not HTTP errors, so that the request can be errored even though the HTTP status code of the response is successful.
```

> Instead, the error has to be captured if the GraphQL response contains an `errors` array, you can do that by Regexing the response body, e.g.:

Reviewer: ```suggestion
Instead, the error has to be captured if the GraphQL response contains an `errors` array. This can be done by regexing the response body, e.g.:
```

```kotlin
val regex = "(?i)\"errors\"\\s*:\\s*\\[".toRegex()

// [body] is the stringified GraphQL response body
if (regex.containsMatchIn(body)) {
    // captures the error
}
```

Additional fields for breadcrumbs:

- data (all fields are optional but recommended):
  - `operation_name` - The GraphQL operation name
  - `operation_type` - The GraphQL operation type, i.e: `query`, `mutation`, `subscription`
  - `operation_id` - The GraphQL operation ID

> Additional fields for the Request interface:

Reviewer: `m`: the `data` field is already part of the specs of Request/Response interface, also the `fingerprints` field is part of the spec of the Event interface. maybe better to word this `Required fields for the xyz interface:` or similar?
